ipn/ipnlocal/peerapi.go

```diff
@@ -776,13 +779,40 @@ func (h *peerAPIHandler) handleDNSQuery(w http.ResponseWriter, r *http.Request)
 		http.Error(w, "DNS access denied", http.StatusForbidden)
 		return
 	}
+	pretty := false // non-DoH debug mode for humans
 	q, publicError := dohQuery(r)
+	if publicError != "" && r.Method == "GET" {
+		if name := r.FormValue("q"); name != "" {
+			pretty = true
+			publicError = ""
+			q = dnsQueryForName(name, r.FormValue("t"))
+		}
+	}
 	if publicError != "" {
 		http.Error(w, publicError, http.StatusBadRequest)
 		return
 	}
-	// TODO(bradfitz): owl.
-	fmt.Fprintf(w, "## TODO: got %d bytes of DNS query", len(q))
+	ctx, cancel := context.WithTimeout(r.Context(), 5*time.Second)
+	defer cancel()
+	res, err := h.ps.resolver.ForwardSync(ctx, q, h.remoteAddr)
```

Comment:
I don't think this is right. We want to hand peerapi queries to the OS for resolution, not to our internal forwarder. Depending on the DNS configuration, it doesn't have the full DNS config.

Reply:
I don't think asking the OS is quite the right level. We don't want them to loop back into us and get our own MagicDNS names. We want "internet only" DNS.

That doesn't quite matter now, but will if/when we allow sharing exit nodes. The person using the exit node will have their DNS set to 100.100.100.100 which will first add any MagicDNS names for their tailnet, but then they'll forward to peerapi DoH which should not add any more MagicDNS names.

But, yeah, I guess we need more here in the case of an exit node that that's using systemd-resolved and doesn't have global/fallback DNS configured.

Reply:
Updated the PR.

Renamed `ForwardSync` to `HandleExitNodeDNSQuery` to better reflect what it's trying to do.

And I handled the systemd-resolved Split DNS case on Linux at least now, even if it not wonderfully. I left some TODOs to clean it up.

My goal here is to get enough working on the peerapi end that I can then work on the client side.